elip-deterministic-blinding-key.mediawiki

```diff
@@ -0,0 +1,122 @@
+  Type: Standards Track
+  Created: 2023-11-08
+  License: BSD-3-Clause
+</pre>
+
+==Introduction==
+
+===Abstract===
+
+This document proposes a standard way to deterministically derive the descriptor blinding key in view descriptors.
+
+===Copyright===
+
+This document is licensed under the 3-clause BSD license.
+
+===Motivation===
+
+CT descriptors as defined in [https://github.com/ElementsProject/ELIPs/blob/main/elip-0150.mediawiki ELIP-150] consist of a descriptor blinding key and an ordinary descriptor.
+
+Descriptor blinding keys can be chosen arbitrarily.
+This however increases the amount of data that the user needs to back up.
+
+Otherwise they can be deterministically derived from a BIP32 seed following [https://github.com/satoshilabs/slips/blob/master/slip-0077.md SLIP-77].
+This works for singlesig wallets,
+since the only backup that the user needs is the BIP32 seed (or BIP39 mnemonic).
+However this does not apply to multisig wallets
+which usually consist of multiple xpubs,
+and thus multiple BIP32 seeds from which a descriptor blinding key could be derived.
+SLIP77 also does not work well when there are multiple BIP44 accounts,
+as they will have the same master blinding key.
+
+This document defines a standard way to deterministically derive a descriptor blinding key from an ordinary descriptor.
+
+Deriving the descriptor blinding key from the ordinary descriptor means that multisig participants can avoid needing a key construction and sharing protocol.
+
+==Design==
+
+===Overview===
+We propose a new possible value for the descriptor blinding key
+
+<pre>
+ct(elipXXXX,<DESCRIPTOR>)
+</pre>
+
+Which indicates that descriptor blinding key to be used is a '''view key''' deterministically derived from the ordinary descriptor <code>DESCRIPTOR</code>
+
+===Drawbacks===
+If a wallet uses this standard to derive its descriptor blinding key, anyone knowing the ordinary descriptor will be able to unblind all the corresponding outputs.
+
+Potential adopters of this standard should consider whether this behavior is acceptable.
+
+===Specification===
+The following descriptor:
+
+<pre>
+ct(elipXXXX,<DESCRIPTOR>)
+</pre>
+
+Is equivalent to:
+
+<pre>
+ct(<KEY>,<DESCRIPTOR>)
+</pre>
+
+Where <code>KEY</code> is a view blinding key derived as follows:
+* If the top-level script expression is <code>combo</code>, fail.
+* Future extensions that support multiple descriptors other than multi-path are not supported.
+* If <code>DESCRIPTOR</code> is multi-path, expand to every single descriptor.
+* For each single descriptor:
+** If it has wildcards, get the definite descriptor at index 2<sup>31</sup>-1<ref>
+We want to use a <code>scriptPubKey</code> that will not be used in a transaction.
+Otherwise it can be used directly by anyone to derive the descriptor blinding key.
+Wallets use <code>scriptPubKey</code>s starting from index 0,
+thus we choose the last non-hardened index,
+which in practice will never be used in a transaction.
```

Comment:
It is already, since it's wrapped in between `<ref></ref>`, see https://github.com/LeoComandini/ELIPs/blob/2023-11-deterministic-blinding-key/elip-deterministic-blinding-key.mediawiki